miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLibraryService.java

```diff
@@ -588,6 +594,20 @@ private void validateChange(Library library, Library beforeChange) throws IOExce
     }
   }
 
+  private void validateTargetedSequencing(LibraryDesignCode libraryDesignCode, Collection<LibraryDilution> libraryDilutions,
+      List<ValidationError> errors) throws IOException {
+    libraryDesignCode = libraryDesignCodeService.get(libraryDesignCode.getId());
+    if (libraryDesignCode.isTargetedSequencingRequired()) {
+      List<String> badDilutions = libraryDilutions.stream().filter(dilution -> dilution.getTargetedSequencing() == null)
+          .map(LibraryDilution::getName).collect(Collectors.toList());
+      if (!badDilutions.isEmpty()) {
+        errors.add(new ValidationError("libraryDesignCodeId",
+            String.format("Library design code %s requires targeted sequencing, but the following dilutions have none: %s",
+                libraryDesignCode.getCode(), String.join(", ", badDilutions))));
```

Comment:
Could the error message be changed to include the action the user needs to take to get it into a non-error state? Maybe something like "Library design code %s requires targeted sequencing. Please add a targeted sequencing to the affected dilutions (%s) then update the library design code." ?

Reply:
"Targeted sequencing must be assigned to the affected dilutions (%s) to use this library design code" for conciseness and consistent wording with other validation messages

Reply:
Done.